labs/b5.md

```diff
@@ -0,0 +1,364 @@
+
+## DNS Records
+DNS servers store data in the form of Resource Records (RR). Resource records are essentially a tuple of (name, value, type, TTL). While there are a wide variety of types of DNS Records the ones we are most concerned with are
+1. **A records**
+name = hostname
+value = IP address
+
+    This record is very simply the record that has the IP address for a given hostname, essentially the information we want to end up with.
+2. **NS records**
+name = domain
+value = name of dns server for domain
+
+    This record points to another dns server that can provide an authoritative  answer for the domain. Think of this as redirecting you to another nameserver.
+3. **CNAME records**
+name = alias
+value = canonical name
+
+    These records point to the canonical name for a given alias for example docs.google.com would be an alias which simply points to documents.google.com
+try ww.facebook.com
+4. **PTR records**
+name = IP address
+value = hostname
+
+    This is essentially the opposite of an A record and is used for reverse DNS lookup, i.e. given an IP address find its corresponding hostname
+
+## DHCP
+The next question we have to answer is exactly how we receive an IP address.  From the perspective of our laptop as a client we turn to Dynamic Host Configuration Protocol (DHCP).
+
+DHCP is a protocol where a DHCP server dynamically distributes network configurations, such as IP addresses, for interfaces (NICs) and other services.
+
+For example, when your machine boots up and connects to AirBears, DHCP assigns your machine a dynamic IP address that uniquely identifies your device within the Berkeley Network.
+![alt text][DHCP1]
+
+[DHCP1]: https://i.imgur.com/Muw9SFf.png "DHCP through Mac GUI"
+
+![alt text][DHCP2]
+
+[DHCP2]: https://i.imgur.com/g5ubmUO.png "DHCP through terminal"
+
+Notice anything special about the IP address in the above examples? [Hint: Look at the network block][private] Additionally, do you recall what must be done so that your unique IP address can communicate over the internet?
+
+To dive a bit deeper into DHCP, this diagram details the typical procedure for a client requesting an IP.
+![alt text][DHCP3]
+
+[DHCP3]: https://upload.wikimedia.org/wikipedia/commons/thumb/e/e4/DHCP_session.svg/260px-DHCP_session.svg.png "DHCP request steps"
+
+The client broadcasts a DHCPDISCOVER message essentially communicating that it wants an IP address. A DHCP server responds to the DHCPDISCOVER with a DHCPOFFER which contains an IP address for the client. If the client accepts the offer, it sends a DHCPREQUEST back to the server. The server acknowledges this by sending a DHCPACK back. The IP address is then considered leased to the machine and is valid for some period of time specified by the DHCP server. Once an IP address lease expires, the client must acquire a new IP address though it does have the option to renew a previous lease for the same IP address.
```

Comment:
backticks around all the special keywords

Reply:
done